File: express/models/product.js
const { Model, DataTypes } = require("sequelize");
const sequelize = require("../config/database");

class Product extends Model {
  static associate(models) {
    Product.belongsToMany(models.Order, {
      through: models.OrderProduct,
      foreignKey: "product_id",
      otherKey: "order_id",
    });

    Product.hasMany(models.OrderProduct, {
      foreignKey: "product_id",
      as: "OrderProducts",
    });
  }
}

Product.init(
  {
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    price: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
    },
    stock: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
  },
  {
    sequelize,
    modelName: "Product",
    tableName: "products",
    timestamps: false,
  }
);

module.exports = Product;

File: asp/performance_asp/performance_asp/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace performance_asp.Models
{
    public class Product
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        [Column("name")]
        public string Name { get; set; }

        [Required]
        [Column("price")]
        public double Price { get; set; }

        [Required]
        [Column("stock")]
        public int Stock { get; set; }

        public virtual ICollection<OrderProduct> OrderProducts { get; set; }
    }
}
